custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -382,6 +339,156 @@ class StockStatus(MultiReport):
     is_exportable = True
 
     @property
+    def get_supply_points(self):
+        supply_points = []
+        if self.report_config['location_id']:
+            location = SQLLocation.objects.get(
+                domain=self.report_config['domain'],
+                location_id=self.report_config['location_id']
+            )
+            if location.location_type.name == 'country':
+                supply_points = SQLLocation.objects.filter(
+                    Q(parent__location_id=self.report_config['location_id'], is_archived=False) |
+                    Q(location_type__name='Regional Medical Store', domain=self.report_config['domain']) |
+                    Q(location_type__name='Teaching Hospital', domain=self.report_config['domain'])
+                ).order_by('name').exclude(supply_point_id__isnull=True)
+            else:
+                supply_points = SQLLocation.objects.filter(
+                    parent__location_id=self.report_config['location_id'], is_archived=False,
+                    location_type__administrative=False,
+                ).order_by('name').exclude(supply_point_id__isnull=True)
+        return supply_points
```

Comment:
This is identical to the `get_supply_points` method used elsewhere, except where `location_id` and `domain` come from.  Can you make a helper function which accepts those two as arguments to avoid this duplication?  Especially  as this looks to be an expensive query.

It also looks like the references to this all call `values_list('supply_point_id', flat=True)` on the result.  Because this isn't memoized, I believe that results in a new query each time.  Regardless, it looks like you could just return the ids from this method and memoize it.